src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -84,6 +96,38 @@ public String updateUserEmail(UserChangeEmailDto userChangeEmailDto, User user)
         return getMessageSource("changed.user.email");
     }
 
+    public String deleteUserFirst(User user) {
+        user.setStatus(DELETED);
+        userRepository.saveAndFlush(user);
+
+        return getParametrizedMessageSource("account.deleted.first", getDaysBeforeDeletion(user));
+    }
+
+    public long getDaysBeforeDeletion(User user) {
+        return numberOfDaysToKeepDeletedUsers - (DAYS.between(user.getUpdated(), LocalDateTime.now()));
+    }
+
+    @Scheduled(cron = ONE_TIME_PER_DAY_AT_3_AM)
+    public void permanentlyDeleteUsers() {
+        userRepository.findAll().stream()
+                .filter(user -> user.getStatus().equals(DELETED) &&
+                        isDurationMoreThanNumberOfDaysToKeepDeletedUser(user.getUpdated()))
+                .forEach(userRepository::delete);
+    }
+
+    private boolean isDurationMoreThanNumberOfDaysToKeepDeletedUser(LocalDateTime localDateTime) {
+        Duration duration = Duration.between(localDateTime, LocalDateTime.now());
+
+        return duration.toDays() > numberOfDaysToKeepDeletedUsers;
+    }
+
+    public String restoreUser(User user) {
+        user.setStatus(ACTIVE);
+        userRepository.saveAndFlush(user);
+
+        return getMessageSource("account.restored");
```

Comment:
You're ruined SOLID's principle "Single responsibility" here - you definitely need to update it.

Reply:
![image](https://user-images.githubusercontent.com/49348448/109987285-2a254880-7d0f-11eb-9b69-98ab5cb0dfb6.png)
Done. But we have the same solution in 2 methods which has been approved recently. Should I fix them? If yes, then within the existing task?